Add ability to copy kubeConfig of current cluster and projects/namespaces to clipboard

jordojordo opened the pull request:
This PR is in reference to #4270 to bring back the ability to copy the kubeconfig for clusters.

Converted the button in the navigation header for downloading the kubeconfig yaml of the current cluster into a dropdown which consists of two options: `Download KubeConfig` and `Copy KubeConfig to Clipboard`

Also added an option in the action-menu of each project/namespace to copy the config to the clipboard.

https://user-images.githubusercontent.com/40806497/135504487-4a71ee75-b92d-4bcf-8c4c-734c870ca754.mp4

nwmac commented on assets/translations/en-us.yaml:
```diff
@@ -2694,6 +2696,7 @@ namespace:
   enableAutoInjection: Enable Istio Auto Injection
   disableAutoInjection: Disable Istio Auto Injection
   move: Move
+  copy: Copy Config to Clipboard
```
Can we name this copyConfig rather than just copy

nwmac commented on assets/translations/en-us.yaml:
```diff
@@ -82,7 +82,9 @@ locale:
   backToRancher: Cluster Manager
   clusterTools: Cluster Tools
-  kubeconfig: Download KubeConfig
+  kubeconfigDownload: Download KubeConfig
```
Since we have may related here, I'd suggest a kubeconfig key with child keys download, copy and options

nwmac commented on components/nav/Header.vue:
```diff
@@ -262,14 +272,51 @@ export default {
 
         <button
-          v-tooltip="t('nav.kubeconfig')"
+          v-tooltip="t('nav.kubeconfigOptions')"
```
The dropdown arrow in the header bar is not obvious as to what it does or the fact its a drop down - I'd suggest we remove the drop down and just put two buttons in the header bar - there's plenty of room and they both have icons.

nwmac reviewed (changes requested):
A few comments - also, this adds a copy config for the namespace resource - not sure why this was added? Doesn't seem to relate to the issue which is to copy the kubeconfig - did I miss something?

jordojordo commented:
Cleaned up the nav header by splitting the dropdown into two separate buttons. Also, removed the copy config action from namespace resources as it didn't relate to the issue.

nwmac commented on components/nav/Header.vue:
```diff
@@ -275,7 +275,17 @@ export default {
         >
-          <i class="icon icon-file icon-lg" />
+          <i class="icon icon-file" />
+        </button>
```
The v-tooltip key need updating - it should be `nav.kubeConfig.download1`

nwmac replied:
@jordojordo I've updated the PR with this fix.
